src/index.js

```diff
@@ -328,44 +396,123 @@ class RedContentScript extends ContentScript {
       this.log('info', 'homePhoneNumber found, inserting it in userIdentity')
       userIdentity.phone.push({
         type: 'home',
-        number: homePhoneNumber.innerHTML
+        number: homePhoneNumber.innerHTML.trim()
       })
     }
+
     await this.sendToPilot({ userIdentity })
   }
 
+  async getContracts() {
+    this.log('info', '📍️ getContracts starts')
+    const contracts = []
+    const actualContractText = document
+      .querySelector(
+        `a[href='https://espace-client.sfr.fr/gestion-ligne/lignes/ajouter']`
+      )
+      .parentNode.parentNode.previousSibling.innerHTML.trim()
+    let actualContractType
+    if (
+      actualContractText.startsWith('06') ||
+      actualContractText.startsWith('07')
+    ) {
+      actualContractType = 'mobile'
+    } else {
+      actualContractType = 'fixe'
+    }
+    const actualContract = {
+      text: actualContractText,
+      id: 'current',
+      type: actualContractType
+    }
+    contracts.push(actualContract)
+    contracts.push(
+      ...Array.from(
+        document
+          .querySelector(
+            `a[href='https://espace-client.sfr.fr/gestion-ligne/lignes/ajouter']`
+          )
+          .parentNode.parentNode.querySelectorAll('li')
+      )
+        .filter(el => !el.getAttribute('class'))
+        .map(el => {
+          const text = el.innerHTML.trim()
+          let type
+          if (text.startsWith('06') || text.startsWith('07')) {
+            type = 'mobile'
+          } else {
+            type = 'fixe'
+          }
+          return {
+            id: el.getAttribute('id') || 'current',
+            text,
+            type
+          }
+        })
+    )
+    return contracts
+  }
+
   async getMoreBills() {
     const moreBillsSelector = 'button[onclick="plusFacture(); return false;"]'
-    while (document.querySelector(`${moreBillsSelector}`) !== null) {
-      this.log('debug', 'moreBillsButton detected, clicking')
-      const moreBillsButton = document.querySelector(`${moreBillsSelector}`)
-      moreBillsButton.click()
-      // Here, we need to wait for the older bills to load on the page
-      await sleep(3)
+    const moreBillAltWrapperSelector = '#plusFacWrap'
+    const moreBillAltSelector = '#plusFac'
+    if (document.querySelector(moreBillsSelector)) {
+      while (document.querySelector(`${moreBillsSelector}`) !== null) {
+        this.log('debug', 'moreBillsButton detected, clicking')
+        const moreBillsButton = document.querySelector(`${moreBillsSelector}`)
+        moreBillsButton.click()
+        // Here, we need to wait for the older bills to load on the page
+        await sleep(3)
```

Comment:
The load of the older bills will take more than 3ms to come. You will miss some bills this way

Reply:
It's done like this since forever, I know that's no excuse but we don't have any other way to ensure it's properly loaded beacause we have no references, no interception, no "numberOfBills" we could find in the HTML nor nothing and sooner in this development, we agreed on using this. What should I do then ?

Reply:
Ah it's not 3ms it's 3s ok. But that is not satisfying. What don't we try to count the number of bills ?

This way we can :

- click on the moreBills button
- the moreBills button vanishes
- we wait for the number of displayed bills to change
- then if the moreBills button is back, we click it

Reply:
This can be done in another PR if needed

Reply:
And im thinking : 

This is not really a problem, if the bills didn't load after 3 sec, we simply will re-click on the moreBillsButton until it disapears as its still there . These behavior leading to nothing but waiting 3 more seconds until the bills finally shows up